app/channel.go

```diff
@@ -411,12 +451,49 @@ func (a *App) createGroupChannel(userIds []string, creatorId string) (*model.Cha
 		Type:        model.CHANNEL_GROUP,
 	}
 
-	channel, err := a.Srv().Store.Channel().Save(group, *a.Config().TeamSettings.MaxChannelsPerTeam)
-	if err != nil {
-		if err.Id == store.CHANNEL_EXISTS_ERROR {
-			return channel, err
+	channel, nErr := a.Srv().Store.Channel().Save(group, *a.Config().TeamSettings.MaxChannelsPerTeam)
+	if nErr != nil {
+		var invErr *store.ErrInvalidInput
+		var intErr *store.ErrInternal
+		var cErr *store.ErrConflict
+		var ltErr *store.ErrLimitExceeded
+		var appErr *model.AppError
+		switch {
+		case errors.As(nErr, &invErr):
+			switch {
+			case invErr.Entity == "Channel" && invErr.Field == "DeleteAt":
+				return nil, model.NewAppError("CreateChannel", "store.sql_channel.save.archived_channel.app_error", nil, "", http.StatusBadRequest)
+			case invErr.Entity == "Channel" && invErr.Field == "Type":
+				return nil, model.NewAppError("CreateChannel", "store.sql_channel.save.direct_channel.app_error", nil, "", http.StatusBadRequest)
+			case invErr.Entity == "Channel" && invErr.Field == "Id":
+				return nil, model.NewAppError("SqlChannelStore.Save", "store.sql_channel.save_channel.existing.app_error", nil, "id="+invErr.Value.(string), http.StatusBadRequest)
+			}
+		case errors.As(nErr, &intErr):
+			switch intErr.Location {
+			case "begin_transaction":
+				return nil, model.NewAppError("CreateChannel", "store.sql_channel.save.open_transaction.app_error", nil, intErr.Error(), http.StatusInternalServerError)
+			case "save_channel":
+				return nil, model.NewAppError("CreateChannel", "store.sql_channel.save_channel.save.app_error", nil, intErr.Error(), http.StatusInternalServerError)
+			case "save_channel_count":
+				return nil, model.NewAppError("CreateChannel", "store.sql_channel.save_channel.current_count.app_error", nil, intErr.Error(), http.StatusInternalServerError)
+			case "upsert_public_channel":
+				return nil, model.NewAppError("CreateChannel", "store.sql_channel.save.upsert_public_channel.app_error", nil, intErr.Error(), http.StatusInternalServerError)
+			case "commit_transaction":
+				return nil, model.NewAppError("SqlChannelStore.Save", "store.sql_channel.save.commit_transaction.app_error", nil, intErr.Error(), http.StatusInternalServerError)
+			}
+		case errors.As(nErr, &cErr):
+			if cErr.Resource == "Channel" {
+				return channel, model.NewAppError("CreateChannel", store.CHANNEL_EXISTS_ERROR, nil, cErr.Error(), http.StatusBadRequest)
+			}
+		case errors.As(nErr, &ltErr):
+			if ltErr.What == "channels_per_team" {
```

Comment:
If we only have one possible limit to exceed there, I would remove the check entirely